Add External Command Line Arguments

HAHOOS opened the pull request:
Adds a new Dictionary variable in `MelonLaunchOptions.cs` called `ExternalArguments` which contains a list of names & values (if found) of command line arguments that are not used by MelonLoader. This can be useful for mods and/or plugins to have the option to use command line arguments. I have tested the changes and it doesn't seem to have any issues. It's not something that will be often used by developers, but I think it would be a great little addition.

HAHOOS committed:
Added External Arguments

HAHOOS committed:
Small change

HAHOOS committed:
Remove summary

HAHOOS committed:
Remove summary, again (for some reason the last commit did not work properly)

HerpDerpinstine commented:
This needs to be rebased to the [alpha-development](/LavaGang/MelonLoader/tree/alpha-development) branch.
Otherwise everything looks good and can be merged in.

HAHOOS commented:
> This needs to be rebased to the [alpha-development](/LavaGang/MelonLoader/tree/alpha-development) branch. Otherwise everything looks good and can be merged in.

Done

HAHOOS committed:
Add summary to ExtendedArguments variable

The summary before did not work correctly, so I removed it, but I've decided to add it back
